core/templates/pages/contributor-dashboard-page/services/machine-translated-text-backend-api.service.spec.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for MachineTranslatedTextBackendApiService.
+*/
+
+import { HttpClientTestingModule, HttpTestingController } from
+  '@angular/common/http/testing';
+import { fakeAsync, flushMicrotasks, TestBed } from '@angular/core/testing';
+
+import { TranslatableItem } from 'domain/opportunity/translatable-content.model';
+import { MachineTranslatedTextBackendApiService, MachineTranslatedTextBackendDict } from './machine-translated-text-backend-api.service';
+
+describe('Machine Translated Texts Backend Api Service', () => {
+  let machineTranslatedTextBackendApiService:
+    MachineTranslatedTextBackendApiService;
+  let httpTestingController: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+    });
+
+    httpTestingController = TestBed.inject(HttpTestingController);
+    machineTranslatedTextBackendApiService = TestBed.inject(
+      MachineTranslatedTextBackendApiService);
+  });
+
+  describe('getMachineTranslatedStateTextsAsync', () => {
+    const sampleTranslationResponse: MachineTranslatedTextBackendDict = {
+      translated_texts: {item1: new TranslatableItem(
+        'texto para traducir', 'html', 'content', null, null)}
+    };
+
+    it('should return a machine generated translation', fakeAsync(() => {
+      machineTranslatedTextBackendApiService
+        .getMachineTranslatedStateTextsAsync(
+          '0', 'Introduction', ['content1'], 'es').then((response) => {
+          expect(response.item1.content).toEqual('texto para traducir');
```

Comment:
Can we refactor this to move the assert statement to the end of the test, e.g. https://github.com/oppia/oppia/blob/2e60d69d7b/core/templates/domain/classroom/classroom-backend-api.service.spec.ts#L77-L96?